ceph.spec.in

```diff
@@ -450,7 +450,8 @@ Requires:      grep
 Requires:      logrotate
 Requires:      parted
 Requires:      psmisc
-Requires:      python%{python3_pkgversion}-setuptools
+Requires:      python%{python3_version_nodots}
```

Comment:
@tchaikov what do you think about this one?

I wasn't sure when I added it: it affects the `base` package. On one hand I didn't want to impose a Ceph-wide dependency on Python 3.8, but on the other hand, it also relies on `python3-setuptools` which will in turn depend on `python3x`, so this makes the dependency more explicit. I personally feel that it's more dangerous to allow coexisting versions of Python on a mono-repo (and assume part of our code base is still 3.6 compliant, without making it clear which ones).

Reply:
yeah, the only python tool packaged by `ceph-base` is the `ceph` cli tool once #41999 gets merged. but `ceph-base` in turn depends on `ceph-common` which packages a bunch of ceph python modules, like

- ceph-argparse
- ceph-common

and python bindings for librados, libcephfs and librbd. all of them are build and installed for a certain python version. so yes, it should depend on a certain python runtime versioned with minor version number.

when it comes to `python3-setuptool`, i moved it to ceph-osd in #41999 as a cleanup. not sure what maintainers of ceph-volume think about the python3.8 dependencies. if we can go further by extracting ceph-volume into its own package, we don't need to worry about this problem anymore. but since `ceph-osd` depends on `ceph-base`, it's non-trivial to make the dependency from ceph-volume to "python3" but not "python3.8" explicit. and as you put, it might not be a best practice to *use* multiple python runtime in a deployment. i concur with you on this, as it'd be a pain for testability and maintainability, even it's completely technically viable and supported by python.

in short, 

- i'd avoid having both `python3-setuptools` and `python38` as runtime dependencies of a single package. 
- if ceph-volume is extracted into its own package, and it does not depend on other ceph python modules. it can depend on `python3-setuptools` or `python38-setuptools`. but if we want to use `python3-setuptools`, we have to test both `python36-setuptools` and `python38-setuptools`.

Reply:
Fixed.

Reply:
split ceph-volume pending on review at #42465

Reply:
BTW, we will have to build python bindings for both python3.8 and python3.6, so we don't break the existing users of these python bindings.